internal/controller/tunnelbinding_controller.go

```diff
@@ -53,12 +53,13 @@ type TunnelBindingReconciler struct {
 
 	// Custom data for ease of (re)use
 
-	ctx            context.Context
-	log            logr.Logger
-	binding        *networkingv1alpha1.TunnelBinding
-	configmap      *corev1.ConfigMap
-	fallbackTarget string
-	cfAPI          *cf.API
+	ctx              context.Context
+	log              logr.Logger
+	binding          *networkingv1alpha1.TunnelBinding
+	configmap        *corev1.ConfigMap
+	fallbackTarget   string
+	cfAPI            *cf.API
+	removedHostnames []string
```

Comment:
Would this have hostnames from all tunnel bindings? I don't particularly like that tbh, where modifying one can result in a cleanup from another binding.

Could we do something per tunnelBinding utilizing status to keep track of previous, and delete them if the spec is changed to a new one?

Reply:
No, it only holds removed hostnames for a particular tunnelBinding.

But the way the code is put together, the `removedHostnames` are shared across tunnelBinding reconciliation. This is fine as long as the `MaxConcurrentReconciles` is not set ([defaults to 1](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller#TypedOptions)), and would get problematic if the concurrency is set larger than 1.